src/Changelog/ReleaseChangelogViaKeepAChangelog.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Changelog;
+
+use Laminas\AutomaticReleases\Git\CommitFile;
+use Laminas\AutomaticReleases\Git\Push;
+use Laminas\AutomaticReleases\Git\Value\BranchName;
+use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
+use Phly\KeepAChangelog\Common\DiscoverChangelogEntryListener;
+use Phly\KeepAChangelog\Config;
+use Phly\KeepAChangelog\Version\ReadyLatestChangelogEvent;
+use Phly\KeepAChangelog\Version\SetDateForChangelogReleaseListener;
+use Symfony\Component\Console\Output\NullOutput;
+
+use function date;
```

Comment:
This should not be: if the current date/time is needed, let's inject it via a `Clock` (https://github.com/lcobucci/clock is a simple abstraction for it)

Reply:
As in, inject it as an argument? as a property injected or created via constructor? Creation of the instance would be something like `new FrozenClock(new DateTimeImmutable('now'))` in the `bin/console.php` script?

Essentially, I just need the date string, so internally, where I have `date('Y-m-d')`, this would be come `$clock->now()->format('Y-m-d')`?

Reply:
in here, `$this->clock->now()->format('Y-m-d')`

In the constructor call, `new SystemClock()` is given as a dependency